easybuild/easyblocks/t/tinker.py

```diff
@@ -0,0 +1,154 @@
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing TINKER, implemented as an easyblock
+
+@author: Kenneth Hoste (Ghent University)
+"""
+import fileinput
+import glob
+import os
+import re
+import shutil
+import sys
+import tempfile
+
+import easybuild.tools.toolchain as toolchain
+from easybuild.framework.easyblock import EasyBlock
+from easybuild.tools.modules import get_software_root
+from easybuild.tools.filetools import mkdir
+from easybuild.tools.run import run_cmd
+from easybuild.tools.systemtools import DARWIN, LINUX, get_os_type
+
+
+class EB_TINKER(EasyBlock):
+    """Support for building/installing TINKER."""
+
+    def __init__(self, *args, **kwargs):
+        """Custom easyblock constructor for TINKER: initialise class variables."""
+        super(EB_TINKER, self).__init__(*args, **kwargs)
+
+        self.build_subdir = None
+        self.build_in_installdir = True
+
+    def configure_step(self):
+        """Custom configuration procedure for TINKER."""
+        # make sure FFTW is available
+        if get_software_root('FFTW') is None:
+            self.log.error("FFTW dependency is not available.")
+
+        os_dirs = {
+            LINUX: 'linux',
+            DARWIN: 'macosx',
+        }
+        os_type = get_os_type()
+        os_dir = os_dirs.get(os_type)
+        if os_dir is None:
+            self.log.error("Failed to determine OS directory for %s (known: %s)" % (os_type, os_dirs))
+
+        comp_dirs = {
+            toolchain.INTELCOMP: 'intel',
+            toolchain.GCC: 'gfortran',
+        }
+        comp_fam = self.toolchain.comp_family()
+        comp_dir = comp_dirs.get(comp_fam)
+        if comp_dir is None:
+            self.log.error("Failed to determine compiler directory for %s (known: %s)" % (comp_fam, comp_dirs))
+
+        self.build_subdir = os.path.join(os_dir, comp_dir)
+        self.log.info("Using build scripts from %s subdirectory" % self.build_subdir)
+
+        # patch 'link.make' script to use FFTW provided via EasyBuild
+        link_make_fp = os.path.join(self.cfg['start_dir'], self.build_subdir, 'link.make')
+        for line in fileinput.input(link_make_fp, inplace=1, backup='.orig'):
+            line = re.sub(r"libfftw3_threads.a libfftw3.a", r"-L$EBROOTFFTW/lib -lfftw3_threads -lfftw3", line)
```

Comment:
it's not a `Makefile`, it's a script, basically

see https://github.com/jayponder/tinker/blob/master/linux/intel/link.make